pkg/controller/builder.go

```diff
@@ -250,3 +253,38 @@ func UpdateDeploymentProxyHook(pc clients.ProxyClientInterface) deploymentcontro
 		return nil
 	}
 }
+
+func UpdateDeploymentFeatureGatesHook(fa featuregates.FeatureGateAccess) deploymentcontroller.DeploymentHookFunc {
+	return func(_ *operatorv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		klog.FromContext(context.Background()).WithName("builder").V(0).Info("FeatureGatesHook updating environment", "deployment", deployment.Name)
+
+		featuregates, err := fa.CurrentFeatureGates()
+		if err != nil {
+			return fmt.Errorf("error getting featuregates.config.openshift.io/cluster: %w", err)
+		}
+
+		vars := []corev1.EnvVar{{Name: "FEATURE_GATES"}}
+		if featuregates.Enabled(features.FeatureGateNewOLM) {
+			vars[0].Value = string(features.FeatureGateNewOLM)
```

Comment:
I wasn't sure to which upstream feature gates we want to have this mapped to, so I'm currently just setting this to the name of the our only one downstream feature gate.
Once the mapping is known, we can adjust and probably lowercase the value(s) here as well.

Additional point - doing the mapping here will introduce fragility with upstream feature gate names, but I am not sure there is a way around it. Deferring it to upstream and pulling an additional openshift dependency isn't probably what we want.

Reply:
It might be good to go ahead and setup a package and/or type that facilitates registering the mapping from downstream feature gates to upstream feature gates. That would help others have a very clear pattern to follow when new downstream feature gates come along.

> doing the mapping here will introduce fragility with upstream feature gate names

I agree there is some fragility here, but downstream seems like the appropriate place to handle the mapping. I would hope that we wouldn't have to deal with this all that often, other than adding new mappings.

Reply:
Can we address some of the fragility by moving the operator-controller/catalogd features package out of internal and importing it directly to figure out all supported upstream feature gates?